Xamarin.Forms.Core/DependencyResolver.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms.Internals
+{
+	public static class DependencyResolver
+	{
+		public delegate object ResolveDelegate(Type type, params object[] args);
+
+		static ResolveDelegate Resolver { get; set; }
+
+		public static void ResolveUsing(ResolveDelegate resolveDelegate)
+		{
+			Resolver = resolveDelegate;
+		}
+
+		public static void ResolveUsing(Func<Type, object> resolveFunc)
+		{
+			object ResolveDelegate(Type type, object[] args) => resolveFunc.Invoke(type);
+			Resolver = ResolveDelegate;
+		}
+
+		internal static object Resolve(Type type, params object[] args)
+		{
+			return Resolver?.Invoke(type, args);
+		}
+
+		internal static object ForceResolve(Type type, params object[] args)
+		{
+			var result = Resolve(type, args);
+
+			if (result != null) return result;
+
+			if (args.Length > 0)
```

Comment:
I'm not sure what case the following code handles; Suggest we add unit tests to cover 35:46.